File: index2.php
<?php

// Function to convert PDF to HTML
function pdfToHtml($pdfPath, $outputDir) {
    // Use the pdftohtml command from Poppler
    $command = "pdftohtml -c \"$pdfPath\" \"$outputDir\"";
    
    // Execute the command and capture output
    exec($command, $output, $returnVar);
    
    // Check if the conversion was successful
    if ($returnVar === 0) {
        // Get the list of HTML files
        $htmlFiles = glob($outputDir . '/*.html');
        
        // Combine the contents of HTML files into one
        $combinedHtmlContent = '';
        foreach ($htmlFiles as $htmlFile) {
            $combinedHtmlContent .= file_get_contents($htmlFile);
        }
        
        // Return the combined HTML content
        return $combinedHtmlContent;
    } else {
        // Return an error message
        return "Conversion failed. Check the error message: " . implode("\n", $output);
    }
}

// Example usage
$pdfFilePath = __DIR__.'/files/JS3.pdf';
$outputDir = __DIR__.'/files/';

// Get combined HTML content from PDF
$htmlContent = pdfToHtml($pdfFilePath, $outputDir);

?>

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>PDF to HTML</title>
</head>
<body>

<?php
// Output the combined HTML content
if (isset($htmlContent)) {
    echo $htmlContent;
} else {
    echo "Error converting PDF to HTML.";
}
?>

</body>
</html>

File: index2.py
import subprocess

def convert_pdf_to_html(pdf_path, output_path):
    try:
        subprocess.run(['pdftohtml', '-noframes', '-enc', 'UTF-8', '-c', '-q', pdf_path, output_path])
        print(f'Successfully converted {pdf_path} to {output_path}')
    except Exception as e:
        print(f'Error converting {pdf_path} to HTML: {e}')

if __name__ == '__main__':
    pdf_file_path = '/Users/mac/Desktop/k2.pdf'
    html_output_path = '/Users/mac/apps/pdf/k4.html'

    convert_pdf_to_html(pdf_file_path, html_output_path)
 

 